src/plugins/vis_type_timelion/server/usage_collector/get_usage_collector.ts

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { SearchResponse } from 'elasticsearch';
+import { ElasticsearchClient } from 'src/core/server';
+import { getIndexArgs } from '../../common/parser';
+import { getIndexPatternsService } from '../services';
+
+type ESResponse = SearchResponse<{ visualization: { visState: string }; updated_at: string }>;
+
+export interface TimelionUsage {
+  timelion_use_scripted_fields_total: number;
+}
+
+const getPastDays = (dateString: string): number => {
+  const date = new Date(dateString);
+  const today = new Date();
+  const diff = Math.abs(date.getTime() - today.getTime());
```

Comment:
maybe momentjs can help us to beautify that part

Reply:
done